ARCHITECTURE.md

```diff
@@ -0,0 +1,110 @@
+
+Actions (e.g., the parsing of CSV files, conversion of media files, or storage of media files in a Pairtree structure, etc.) are performed by the application's various verticles (e.g., WatcherVerticle, ConverterVerticle, PairtreeVerticle, etc.) Cf. the `verticles` directory for examples.
+
+## Sequence Diagram
+
+The sequence diagram below illustrates the order of events that happen as the av-pairtree program processes an A/V file and puts it into a [Pairtree](https://tools.ietf.org/html/draft-kunze-pairtree-01) directory structure.
+
+![Sequence diagram for av-pairtree's workflow](docs/images/av_pairtree_sequence.svg)
+
+A single CSV file might contain both audio and media files, so av-pairtree loops through both processes in the sequence diagram as it parses content from the supplied CSV file. It also loops through CSV submissions, since more than one CSV can be submitted via the drop box.
+
+## Configuration Options
+
+The sample configuration file for the project can be found in the project's test resources directory: `src/test/resources/test-config.properties`. This configuration is also what's used when running the project's tests.
+
+When running av-pairtree in a non-test mode, one needs to supply their own configuration file, as described in the project's `README.md`.
+
+The properties and default values for an av-pairtree configuration file are as follows:
+
+| Property | Description | Default value |
+--- | --- | ---
+| http.port | The port at which the HTTP server runs | 8888 |
+| http.host | The host name used by the HTTP server | 0.0.0.0 |
+| csv.dir | The watched directory into which CSV are dropped | N/A |
+| source.dir | The root directory for the CSV's media files | N/A |
+| output.dir | The place where a collection's Pairtrees are written | N/A |
+| pairtree.prefix | The optional Pairtree prefix that should be used when creating Pairtrees | N/A |
+| audio.encoding.format | The output audio encoding format | mp4 |
+| audio.sampling.rate | The output audio sampling rate | 44100 |
+| audio.codec | The audio codec to use inside the encoding wrapper | aac |
+| audio.bit.rate | The bit rate used to encoding the audio stream | 128000 |
+| audio.channels | The number of channels in the audio stream | 2 |
+| iiif.access.url | The URL pattern into which to insert the Pairtree path | N/A |
+| conversion.workers | The number of cores to use for audio file conversion | 2 |
+
+## Documentation
+
+This project has a `README.md` file and an `ARCHITECTURE.md` file (what you're currently reading) in its root directory.
+
+The application also uses OpenAPI and AsyncAPI files to define its public and internal messaging APIs. These can be found in the [`src/main/resources`](https://github.com/UCLALibrary/av-pairtree/tree/main/src/main/resources) directory. The files are `av-pairtree-openapi.yaml` and `av-pairtree-asyncapi.yaml`.
```

Comment:
Just flagging that there isn't an AsyncAPI doc yet, but I assume it may be added soon?

Reply:
Yes, the file name is just in anticipation of there needing to be another yaml file in that same dir with basically the same name.